nav2_docking/opennav_docking/src/docking_server.cpp

```diff
@@ -447,7 +509,7 @@ bool DockingServer::approachDock(
     }
 
     // Update perception
-    if (!dock->plugin->getRefinedPose(dock_pose, dock->id)) {
+    if (!dock->plugin->shouldRotateToDock() && !dock->plugin->getRefinedPose(dock_pose, dock->id)) {
```

Comment:
Can we invert this to be `!dock->plugin->getRefinedPose(dock_pose, dock->id) && !dock->plugin->shouldRotateToDock()`? 

I think that would enable us to get new refined poses when backing up when they are available (if not blind). If we do the opposite, then we check if rotating and if so, we so skip the second check which does the update.